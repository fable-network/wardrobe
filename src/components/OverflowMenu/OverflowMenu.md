Basic usage:
```jsx
<OverflowMenu>
  <OverflowMenu.Item>
    Edit
  </OverflowMenu.Item>
  <OverflowMenu.Item onClick={() => alert("this.delete()")}>
    Delete
  </OverflowMenu.Item>
</OverflowMenu>
```

Different position:
```jsx
<OverflowMenu position="left">
  <OverflowMenu.Item>
    Edit
  </OverflowMenu.Item>
  <OverflowMenu.Item>
    Delete
  </OverflowMenu.Item>
</OverflowMenu>
```

Custom colors:
```jsx
<OverflowMenu color='red' activeColor='orange'>
  <OverflowMenu.Item>
    Action 1
  </OverflowMenu.Item>
  <OverflowMenu.Item>
    Action 2
  </OverflowMenu.Item>
</OverflowMenu>
```

Open initially (with title):
```jsx
<OverflowMenu openByDefault={true}>
  <OverflowMenu.Title>Use the trigger to close me</OverflowMenu.Title>
  <OverflowMenu.Item>
    Action 1
  </OverflowMenu.Item>
  <OverflowMenu.Item>
    Action 2
  </OverflowMenu.Item>
</OverflowMenu>
```
