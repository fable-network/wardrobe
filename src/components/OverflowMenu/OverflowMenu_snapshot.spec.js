/* eslint-disable */
// This file was automatically generated
import React from 'react';
import { shallow } from 'enzyme';
import OverflowMenu from './OverflowMenu';

describe('OverflowMenu Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = shallow(<OverflowMenu> <OverflowMenu.Item onClick={() => alert("this.edit()")}> Edit </OverflowMenu.Item> <OverflowMenu.Item onClick={() => alert("this.delete()")}> Delete </OverflowMenu.Item></OverflowMenu>);
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = shallow(<OverflowMenu position="left"> <OverflowMenu.Item> Edit </OverflowMenu.Item> <OverflowMenu.Item> Delete </OverflowMenu.Item></OverflowMenu>);
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = shallow(<OverflowMenu appearance="primary"> <OverflowMenu.Item> Edit </OverflowMenu.Item> <OverflowMenu.Item> Delete </OverflowMenu.Item></OverflowMenu>);
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = shallow(<OverflowMenu appearance="secondary"> <OverflowMenu.Item> Edit </OverflowMenu.Item> <OverflowMenu.Item> Delete </OverflowMenu.Item></OverflowMenu>);
      expect(wrapper4).toMatchSnapshot();

      const wrapper5 = shallow(<OverflowMenu color='red'> <OverflowMenu.Item> Action 1 </OverflowMenu.Item> <OverflowMenu.Item> Action 2 </OverflowMenu.Item></OverflowMenu>);
      expect(wrapper5).toMatchSnapshot();

      const wrapper6 = shallow(<OverflowMenu openByDefault={true}> <OverflowMenu.Title>Use the trigger to close me</OverflowMenu.Title> <OverflowMenu.Item> Action 1 </OverflowMenu.Item> <OverflowMenu.Item> Action 2 </OverflowMenu.Item></OverflowMenu>);
      expect(wrapper6).toMatchSnapshot();

      const wrapper7 = shallow(<OverflowMenu openByDefault={true} appearance="primary"> <OverflowMenu.Item> Action 1 </OverflowMenu.Item> <OverflowMenu.Item> Action 2 </OverflowMenu.Item></OverflowMenu>);
      expect(wrapper7).toMatchSnapshot();

    });
  });
});
