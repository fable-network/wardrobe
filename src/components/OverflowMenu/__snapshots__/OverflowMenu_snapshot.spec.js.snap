// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverflowMenu Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(OverflowMenu)>
    <DropdownItem
      onClick={[Function]}
    >
      Edit
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      Delete
    </DropdownItem>
  </WithTheme(OverflowMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appearance": "dark",
      "children": Array [
        <DropdownItem
          onClick={[Function]}
        >
          Edit
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Delete
        </DropdownItem>,
      ],
      "onClose": [Function],
      "onOpen": [Function],
      "position": "right",
      "size": "small",
      "theme": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Edit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Edit",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appearance": "dark",
        "children": Array [
          <DropdownItem
            onClick={[Function]}
          >
            Edit
          </DropdownItem>,
          <DropdownItem
            onClick={[Function]}
          >
            Delete
          </DropdownItem>,
        ],
        "onClose": [Function],
        "onOpen": [Function],
        "position": "right",
        "size": "small",
        "theme": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Edit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Edit",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`OverflowMenu Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(OverflowMenu)
    size="medium"
  >
    <DropdownItem
      onClick={[Function]}
    >
      Edit
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      Delete
    </DropdownItem>
  </WithTheme(OverflowMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appearance": "dark",
      "children": Array [
        <DropdownItem
          onClick={[Function]}
        >
          Edit
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Delete
        </DropdownItem>,
      ],
      "onClose": [Function],
      "onOpen": [Function],
      "position": "right",
      "size": "medium",
      "theme": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Edit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Edit",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appearance": "dark",
        "children": Array [
          <DropdownItem
            onClick={[Function]}
          >
            Edit
          </DropdownItem>,
          <DropdownItem
            onClick={[Function]}
          >
            Delete
          </DropdownItem>,
        ],
        "onClose": [Function],
        "onOpen": [Function],
        "position": "right",
        "size": "medium",
        "theme": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Edit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Edit",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`OverflowMenu Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(OverflowMenu)
    size="large"
  >
    <DropdownItem
      onClick={[Function]}
    >
      Edit
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      Delete
    </DropdownItem>
  </WithTheme(OverflowMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appearance": "dark",
      "children": Array [
        <DropdownItem
          onClick={[Function]}
        >
          Edit
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Delete
        </DropdownItem>,
      ],
      "onClose": [Function],
      "onOpen": [Function],
      "position": "right",
      "size": "large",
      "theme": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Edit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Edit",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appearance": "dark",
        "children": Array [
          <DropdownItem
            onClick={[Function]}
          >
            Edit
          </DropdownItem>,
          <DropdownItem
            onClick={[Function]}
          >
            Delete
          </DropdownItem>,
        ],
        "onClose": [Function],
        "onOpen": [Function],
        "position": "right",
        "size": "large",
        "theme": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Edit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Edit",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`OverflowMenu Component snapshots should match snapshot 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(OverflowMenu)
    position="left"
  >
    <DropdownItem
      onClick={[Function]}
    >
      Edit
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      Delete
    </DropdownItem>
  </WithTheme(OverflowMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appearance": "dark",
      "children": Array [
        <DropdownItem
          onClick={[Function]}
        >
          Edit
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Delete
        </DropdownItem>,
      ],
      "onClose": [Function],
      "onOpen": [Function],
      "position": "left",
      "size": "small",
      "theme": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Edit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Edit",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appearance": "dark",
        "children": Array [
          <DropdownItem
            onClick={[Function]}
          >
            Edit
          </DropdownItem>,
          <DropdownItem
            onClick={[Function]}
          >
            Delete
          </DropdownItem>,
        ],
        "onClose": [Function],
        "onOpen": [Function],
        "position": "left",
        "size": "small",
        "theme": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Edit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Edit",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`OverflowMenu Component snapshots should match snapshot 5`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(OverflowMenu)
    appearance="primary"
  >
    <DropdownItem
      onClick={[Function]}
    >
      Edit
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      Delete
    </DropdownItem>
  </WithTheme(OverflowMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appearance": "primary",
      "children": Array [
        <DropdownItem
          onClick={[Function]}
        >
          Edit
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Delete
        </DropdownItem>,
      ],
      "onClose": [Function],
      "onOpen": [Function],
      "position": "right",
      "size": "small",
      "theme": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Edit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Edit",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appearance": "primary",
        "children": Array [
          <DropdownItem
            onClick={[Function]}
          >
            Edit
          </DropdownItem>,
          <DropdownItem
            onClick={[Function]}
          >
            Delete
          </DropdownItem>,
        ],
        "onClose": [Function],
        "onOpen": [Function],
        "position": "right",
        "size": "small",
        "theme": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Edit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Edit",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`OverflowMenu Component snapshots should match snapshot 6`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(OverflowMenu)
    appearance="secondary"
  >
    <DropdownItem
      onClick={[Function]}
    >
      Edit
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      Delete
    </DropdownItem>
  </WithTheme(OverflowMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appearance": "secondary",
      "children": Array [
        <DropdownItem
          onClick={[Function]}
        >
          Edit
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Delete
        </DropdownItem>,
      ],
      "onClose": [Function],
      "onOpen": [Function],
      "position": "right",
      "size": "small",
      "theme": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Edit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Edit",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appearance": "secondary",
        "children": Array [
          <DropdownItem
            onClick={[Function]}
          >
            Edit
          </DropdownItem>,
          <DropdownItem
            onClick={[Function]}
          >
            Delete
          </DropdownItem>,
        ],
        "onClose": [Function],
        "onOpen": [Function],
        "position": "right",
        "size": "small",
        "theme": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Edit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Edit",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`OverflowMenu Component snapshots should match snapshot 7`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(OverflowMenu)
    color="red"
  >
    <DropdownItem
      onClick={[Function]}
    >
      Action 1
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      Action 2
    </DropdownItem>
  </WithTheme(OverflowMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appearance": "dark",
      "children": Array [
        <DropdownItem
          onClick={[Function]}
        >
          Action 1
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Action 2
        </DropdownItem>,
      ],
      "color": "red",
      "onClose": [Function],
      "onOpen": [Function],
      "position": "right",
      "size": "small",
      "theme": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Action 1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Action 1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Action 2",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Action 2",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appearance": "dark",
        "children": Array [
          <DropdownItem
            onClick={[Function]}
          >
            Action 1
          </DropdownItem>,
          <DropdownItem
            onClick={[Function]}
          >
            Action 2
          </DropdownItem>,
        ],
        "color": "red",
        "onClose": [Function],
        "onOpen": [Function],
        "position": "right",
        "size": "small",
        "theme": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Action 1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Action 1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Action 2",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Action 2",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`OverflowMenu Component snapshots should match snapshot 8`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(OverflowMenu)
    activeColor="blue"
  >
    <DropdownItem
      onClick={[Function]}
    >
      Action 1
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      Action 2
    </DropdownItem>
  </WithTheme(OverflowMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeColor": "blue",
      "appearance": "dark",
      "children": Array [
        <DropdownItem
          onClick={[Function]}
        >
          Action 1
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Action 2
        </DropdownItem>,
      ],
      "onClose": [Function],
      "onOpen": [Function],
      "position": "right",
      "size": "small",
      "theme": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Action 1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Action 1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Action 2",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Action 2",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeColor": "blue",
        "appearance": "dark",
        "children": Array [
          <DropdownItem
            onClick={[Function]}
          >
            Action 1
          </DropdownItem>,
          <DropdownItem
            onClick={[Function]}
          >
            Action 2
          </DropdownItem>,
        ],
        "onClose": [Function],
        "onOpen": [Function],
        "position": "right",
        "size": "small",
        "theme": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Action 1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Action 1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Action 2",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Action 2",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`OverflowMenu Component snapshots should match snapshot 9`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithTheme(OverflowMenu)
    open={true}
  >
    <DropdownTitle>
      You can't close me
    </DropdownTitle>
    <DropdownItem
      onClick={[Function]}
    >
      Action 1
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      Action 2
    </DropdownItem>
  </WithTheme(OverflowMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "appearance": "dark",
      "children": Array [
        <DropdownTitle>
          You can't close me
        </DropdownTitle>,
        <DropdownItem
          onClick={[Function]}
        >
          Action 1
        </DropdownItem>,
        <DropdownItem
          onClick={[Function]}
        >
          Action 2
        </DropdownItem>,
      ],
      "onClose": [Function],
      "onOpen": [Function],
      "open": true,
      "position": "right",
      "size": "small",
      "theme": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "You can't close me",
        },
        "ref": null,
        "rendered": "You can't close me",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Action 1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Action 1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Action 2",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Action 2",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appearance": "dark",
        "children": Array [
          <DropdownTitle>
            You can't close me
          </DropdownTitle>,
          <DropdownItem
            onClick={[Function]}
          >
            Action 1
          </DropdownItem>,
          <DropdownItem
            onClick={[Function]}
          >
            Action 2
          </DropdownItem>,
        ],
        "onClose": [Function],
        "onOpen": [Function],
        "open": true,
        "position": "right",
        "size": "small",
        "theme": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "You can't close me",
          },
          "ref": null,
          "rendered": "You can't close me",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Action 1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Action 1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Action 2",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Action 2",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
